packages/wallet-backend-update/src/account/service.ts

```diff
@@ -198,6 +201,68 @@ export class AccountService implements IAccountService {
     }
   }
 
+  public async withdrawFunds(args: FundAccountArgs): Promise<void> {
+    const existingAccount = await Account.query()
+      .where('userId', args.userId)
+      .where('assetCode', args.assetCode)
+      .first()
+    if (!existingAccount) {
+      throw new NotFound()
+    }
+
+    // get list of payout method types for currency, if we get to production: get payout type required fields will be needed
+    const payoutType = await this.deps.rapyd.getPayoutMethodTypes(
+      args.assetCode
+    )
+
+    if (payoutType.status.status !== 'SUCCESS') {
+      throw new Error(
+        `Unable to withdraw funds from your account: ${payoutType.status.message}`
+      )
+    }
+
+    const user = await User.query().findById(args.userId)
+    if (!user || !user.rapydWalletId) {
+      throw new NotFound()
+    }
+
+    // withdraw funds/create payout from wallet account into bank account
+    const userDetails = {
+      name: `${user.firstName} ${user.lastName}`,
+      address: user.address ?? ''
+    }
+    const payout = await this.deps.rapyd.withdrawFundsFromAccount({
+      beneficiary: userDetails,
+      payout_amount: args.amount,
+      payout_currency: args.assetCode,
+      ewallet: user.rapydWalletId ?? '',
+      sender: userDetails,
+      sender_country: user.country ?? '',
+      sender_currency: args.assetCode,
+      beneficiary_entity_type: 'individual',
+      sender_entity_type: 'individual',
+      payout_method_type: payoutType.data[0].payout_method_type
+    })
+
+    if (payout.status.status !== 'SUCCESS') {
+      throw new Error(
+        `Unable to withdraw funds from your account: ${payout.status.message}`
+      )
+    }
+
+    // complete third party/bank payout
+    const completePayoutResponse = await this.deps.rapyd.completePayout({
+      payout: payout.data.id,
+      amount: args.amount
+    })
+
+    if (completePayoutResponse.status.status !== 'SUCCESS') {
+      throw new Error(
+        `Unable to withdraw funds from your account: ${completePayoutResponse.status.message}`
+      )
+    }
```

Comment:
Should we also create a transaction when the user withdraws money?

Reply:
The we need a transaction for funding as well, I think I will create a new stroy on this in Tech Debt, to add funding and withdrawing in transactions list